gourmet/gtk/recipe.py

```diff
@@ -0,0 +1,787 @@
+        builder.get_object('recImage').set_property('no-show-all', True)
+        builder.get_object('delImageButton').set_property('visible', False)
+        builder.get_object('delImageButton').set_property('no-show-all', True)
+        image = _ImageData()
+        builder.connect_signals({
+            'setRecImage': lambda _: _select_image(
+                image,
+                builder.get_object('recImage'),
+                builder.get_object('addImage'),
+                builder.get_object('delImageButton')
+            ),
+            'delRecImage': lambda _: _remove_image(
+                image,
+                builder.get_object('recImage'),
+                builder.get_object('addImage'),
+                builder.get_object('delImageButton')
+            ),
+            })
+
+        notebook = _Gtk.Notebook(tab_pos=_Gtk.PositionType.LEFT)
+
+        # Setup Description tab
+        _configure_combobox(builder, 'categoryBox', rec_gui, 'category')
+        _configure_combobox(builder, 'cuisineBox', rec_gui, 'cuisine')
+        _configure_combobox(builder, 'sourceBox', rec_gui, 'source')
+        _configure_combobox(builder, 'yield_unitBox', rec_gui, 'yield_unit')
+        # TODO: Don't put the main widget inside a parent
+        widget = builder.get_object('vbox26')
+        widget.unparent()
+        notebook.append_page(widget, _Gtk.Label(label=_('Description')))
+
+
+        # Setup Ingredients tab
+        ingredients = _create_ingredients_page()
+        add_ingredient = lambda _: _add_ingredient(ingredients, rec_gui)
+        ingredients.ingredient_entry.connect('activate', add_ingredient)
+        ingredients.ingredient_button.connect('clicked', add_ingredient)
+        add_group = lambda _: _add_group(dialog, ingredients)
+        ingredients.group_button.connect('clicked', add_group)
+
+        deleted_ingredients = []
+        _configure_ingredients_tree(
+            ingredients.ingredients_tree,
+            ingredients.group_combobox,
+            rec_gui,
+            deleted_ingredients)
+        notebook.append_page(ingredients.main, _Gtk.Label(label=_('Ingredients')))
+
+        # Setup Instructions tab
+        instructions = _Gtk.TextView.new_with_buffer(_PangoBuffer())
+        instructions.set_wrap_mode(_Gtk.WrapMode.WORD)
+        notebook.append_page(instructions, _Gtk.Label(label=_('Instructions')))
+
+        # Setup Notes tab
+        notes = _Gtk.TextView.new_with_buffer(_PangoBuffer())
+        notes.set_wrap_mode(_Gtk.WrapMode.WORD)
+        notebook.append_page(notes, _Gtk.Label(label=_('Notes')))
+
+        content = dialog.get_content_area()
+        content.pack_start(notebook, True, True, 0)
+
+        if recipe:
+            # TODO: Elminate useless try block?
+            # This uses the same logic as the original code, but should this
+            # ever fail?
+            try:
+                yields = float(recipe['yields'])
```

Comment:
Yes, the try/except can be removed.

Reply:
Are we sure? `yields` might be `None`, which could raise an error. The TODO is meant as a reminder/placeholder to take a closer look at this.